content/ADR-237-ui-custom-components.md

```diff
@@ -0,0 +1,144 @@
+
+2. **Composability**: Components that return JSX.Elements can be composed together to create complex UIs. This is a key feature of React and should be leveraged in our library.
+
+3. **Interoperability**: Components that return JSX.Elements can be used seamlessly with other React components. This increases the utility of our library and makes it easier for developers to integrate it into their projects.
+
+## Implications
+
+This decision will have the following implications:
+
+1. **Refactoring**: The external library will need to be refactored to comply with this rule. This will involve some effort, but it is a necessary step to ensure consistency and composability across our codebase.
+
+2. **Documentation**: Our documentation will need to be updated to clearly state that all components must return JSX.Elements and to explain the theme system. This will provide clarity to developers on our design decisions and how to use our library correctly.
+
+3. **Guidance**: We may need to provide guidance or resources to developers who are not familiar with this pattern. This could include examples, tutorials, or other educational resources.
+
+4. **Validation**: We will need to implement a mechanism to validate that all components return JSX.Elements. This could be done through static type checking, unit tests, or other methods.
+
+## Links
+
+- [React UI ADR](https://adr.decentraland.org/adr/ADR-124)
+- [React docs](https://react.dev)
+- [React State changes](https://react.dev/learn/reacting-to-input-with-state)
+
+## Theme System Implementation
+
+Here's how we propose to implement a light and dark theme system:
+
+First, create a `ThemeContext`:
+
+```tsx
+import ReactEcs from '@dcl/sdk/react-ecs'
+import React from 'react'
+
+export type Theme = 'light' | 'dark'
+export type ThemeContextProps = {
+  theme: Theme
+  toggleTheme: () => void
+}
+
+export const ThemeContext = React.createContext<ThemeContextProps | undefined>(undefined)
+```
+
+Next, create a `ThemeProvider` component that will provide the theme to its children:
+
+```tsx
+import ReactEcs from '@dcl/sdk/react-ecs'
+import React from 'react'
+import { Theme, ThemeContext } from './ThemeContext'
+
+export function ThemeProvider({ children }: React.PropsWithChildren<{}>) {
+  const [theme, setTheme] = React.useState<Theme>('light')s
+
+  const toggleTheme = () => {
+    setTheme(theme === 'light' ? 'dark' : 'light')
+  }
+
+  return (
+    <ThemeContext.Provider value={{ theme, toggleTheme }}>
+      {children}
+    </ThemeContext.Provider>
+  )
+}
+
+export default ThemeProvider
+```
+
+Now, you can use the `useContext` hook in any functional component to access the current theme and the `toggleTheme` function:
+
+```tsx
+import ReactEcs, { UiEntity } from '@dcl/sdk/react-ecs'
+import React from 'react'
+import { Color4 } from '@dcl/sdk/math'
+
+import { ThemeContext } from './ThemeContext'
+
+const Button = () => {
```

Comment:
Since this will be used as reference, I'd add the `React.FC` types here and wherever possible

Reply:
I think i have to export our own FC types because they conflict with the ones with react 😞